### ISSUE TITLE
[Tutorials] Normalizing the tutorials

### ISSUE THREAD

**Issue by dcfidalgo** [recognai/rubrix#716]
As discussed in slack I did a quick preliminary analysis.

### Data set sizes

These are the tutorials and the data sets used in each one of them, plus their sizes.
- *(< possible)* means that we can easily reduce the size (i would suggest via the datasets streaming feature) without compromising the tutorial
- *(< not possible)* means that a reduction of the data set would alter the results/purpose of the tutorial.

In my opinion, up to 30MB for a data set is fine. If we set the limit at 30MB, then no tutorial needs substantial change regarding the data set size.

##### How to label your data and fine-tune a sentiment classifier
- banking77: 1MB (< not possible)
##### Building a news classifier with weak supervision
  - ag_news: 30MB (< not possible)
##### Explore and analyze Spacy NER pipelines
  - gutenberg_time: 34MB (< possible)
  - imdb: 80MB (< possible)
##### Active learning with ModAL and scikit-learn
  - data shipped
##### Find label errors with cleanlab
  - mrpc: 1.5MB (< not possible)
##### Zero-shot Named Entity Recognition with Flair
  - wnut: <1MB (< possible)
##### Clean labels using your model loss
  - ag_news: 30MB (< not possible)
##### Monitor predictions in HTTP API endpoints
  - no data set used
##### Faster data annotation with a zero-shot
  - mlsum: 490MB (< possible)

### Normalizing structure

Here is my proposal for normalizing the tutorials:

- Heading: Title
- below the title a TDLR (without any extra heading), I imagine one or two summarizing phrases plus a bullet list of the tutorial's steps
- Picture/Diagram/Video
- Heading: Introduction (Optional, describing a bit the topic of the tutorial)
- Heading: Setup (Here, I would link to our GitHub repo, our Rubrix setup guide, install the tutorial's dependencies, do all the imports)
- The actual tutorial. Personally, I like commenting (nearly) every line of the code. When you look at our tutorials, the code snippets with their dark theme are very eye-catching, so often I find myself scrolling through the tutorial only looking at the code snippets (actually I often do the same for other blog posts/tutorials). So having a concise one-line comment is quite valuable in my oppinion.
- Heading: Summary (Optional, but I highly recommend it)
- Heading: Next steps (providing links to further resources)

There is still the issue of how to keep Screenshots/Videos up to date with our new releases, but not sure if we can fully automatize this ... regarding this issue, maybe I would prefer to put each tutorial in its separate folder together with all its media files, so it's very obvious what needs to be updated.

@dvsrepo @frascuchon

**Comment by frascuchon** [comment]
+1 to this:

>  ...maybe I would prefer to put each tutorial in its separate folder together with all its media files, so it's very obvious what needs to be updated.

For the rest of points. We can try the  datasets streaming feature and for large datasets or datasets with inestable connections, we could maintain an alternative dataset version in S3 for example

**Comment by dvsrepo** [comment]
This one can go away: Faster data annotation with a zero-shot

**Comment by dvsrepo** [comment]
The structure is fine, I'd add the author

**Comment by dcfidalgo** [comment]
Tutorials are now pretty much normalized regarding their structure.
The next step should be to update all the media in the tutorials, and maybe some improvements of the tutorials themselves.
